### PR TITLE
Change permissions to match salt for shared directories.

### PR DESCRIPTION
**Pull request by oms4suse**
In addition cleaned up globing of directories.

Signed-off-by: Owen Synge <osynge@suse.com>

### CODE REVIEW

File: deepsea.spec [hunk 1]
```diff
@@ -273,66 +272,90 @@ sed -i '/^master_minion:/s!_REPLACE_ME_!'`hostname -f`'!' /srv/pillar/ceph/maste
 %postun 
 
 %files
-%defattr(-,salt,salt,-)
-/srv/modules/pillar/*
-%dir /etc/salt
-%dir /etc/salt/master.d
+%defattr(-,root,root,-)
+/srv/modules/pillar/stack.py
+%dir %attr(0750, root, salt) %_sysconfdir/salt
```
Comment (l-mb) [inline deepsea.spec]: Aren't these directories owned by the salt package already?

File: deepsea.spec [hunk 2]
```diff
@@ -273,66 +272,90 @@ sed -i '/^master_minion:/s!_REPLACE_ME_!'`hostname -f`'!' /srv/pillar/ceph/maste
...
+%dir %attr(0755, root, salt) %{_sysconfdir}/salt/master.d/
 %dir /%{_runners}
 %dir /%{_pillar}
 %dir /%{_pillar}/ceph
 %dir /%{_pillar}/ceph/stack
 %dir /srv/modules
 %dir /srv/modules/pillar
-%dir /srv/salt
 %dir /srv/salt/_modules
 %dir /%{_saltceph}
```
Comment (l-mb) [inline deepsea.spec]: Shouldn't we specify the attr here too?

File: deepsea.spec [hunk 3]
```diff
@@ -273,66 +272,90 @@ sed -i '/^master_minion:/s!_REPLACE_ME_!'`hostname -f`'!' /srv/pillar/ceph/maste
...
-%config /%{_saltceph}/reactor/*
-%config /%{_saltceph}/refresh/*
-%config /%{_saltceph}/repo/*
-%config /%{_saltceph}/rgw/*
-%config /%{_saltceph}/stage/*
-%config /%{_saltceph}/sync/*
-%config /%{_saltceph}/time/*
-%config /%{_saltceph}/updates/*
+%config /srv/salt/_modules/*.py
+%config /%{_saltceph}/admin/*.sls
```
Comment (l-mb) [inline deepsea.spec]: If we included /%saltceph/ w/o the %dir directive, wouldn't that automatically pull in all subdirectories too?

File: deepsea.spec [hunk 4]
```diff
@@ -273,66 +272,90 @@ sed -i '/^master_minion:/s!_REPLACE_ME_!'`hostname -f`'!' /srv/pillar/ceph/maste
...
+%config /%{_saltceph}/repo/*.sls
+%config /%{_saltceph}/rgw/*.sls
+%config /%{_saltceph}/stage/*.sls
+%config /%{_saltceph}/stage/configure/*.sls
+%config /%{_saltceph}/stage/discovery/*.sls
+%config /%{_saltceph}/stage/prep/*.sls
+%config /%{_saltceph}/sync/*.sls
+%config /%{_saltceph}/time/*.sls
+%config /%{_saltceph}/updates/*.sls
 %doc
```
Comment (l-mb) [inline deepsea.spec]: I think you may want %docdir here?

File: deepsea.spec (hunk 2 above)
Comment (oms4suse) [inline deepsea.spec]: %defattr(-,root,root,-)

matches the salt packaging.

File: deepsea.spec (hunk 3 above)
Comment (oms4suse) [inline deepsea.spec]: This is not a good plan as then the only rpm that can ever install content in %{_saltceph}.

It is also recommended as good practice to use the %dir in many rpm packaging guides.

File: deepsea.spec (hunk 4 above)
Comment (oms4suse) [inline deepsea.spec]: I tend to agree here, but this is an enhancement that could lead to delay in getting the patch accepted.

File: deepsea.spec (hunk 1 above)
Comment (oms4suse) [inline deepsea.spec]: I believe they are, and hope we can resolve this better with a future bug report and patch.

**Pull request merged**